### DISCUSSION TITLE
Power Meter - TASMOTA (HICHI) auf Real-Time ändern anstatt Teleperiod Zeit von 10-300 Sek

### DISCUSSION THREAD

**Discussion by Manos1966** [hoylabs/OpenDTU-OnBattery#412] [General]
Rein Zufällig fand ich etwas interessantes
https://sites.google.com/view/hichi-lesekopf/startseite

MQTT sendet zu oft oder zu selten:
In den Tasmota Loggin Einstellungen kannst du die Teleperiod Zeit zwischen 10-300 Sekunden stellen.
Als Standart steht 300 drin. Daher sendet Tasmota alle 5 Minuten wenn man es nicht ändert.

**Wenn du häufiger als 10 Sekunden senden möchtest musst du die descriptor ändern,**
**Dann wird bei jeder Werte Änderung gesendet**
Aus dem Tasmota Wiki:
1,77070100100700ff@1,P,W,P,18  
_the last entry is the number of digits + an optional flag to send the value as fast as possible code = 16 so 16+2 = 18 means 2 digits + fast send_

Beispiel:
Wenn deine Zeile so aussieht: 
1,77070100100700ff@1,P,W,P,0 
rechnest du 0+16=16 und änderst es so:
1,77070100100700ff@1,P,W,P,16

Also immer die letzte Zahlt in Descriptor plus oder minus 16.

Alles hier Beschrieben: https://tasmota.github.io/docs/Smart-Meter-Interface/#meter-metrics

**Comment by madmartin** [comment]
Danke @Manos1966 ,

bitte ab ins Wiki mit solchen Hinweisen.

Mein ganz persönlicher Kommentar dazu:
Mein Stromzähler sendet alle 1 oder 2 sekunden neue Werte, und der DPL braucht aber nicht so oft Werte. 
Diese Einstellung ist sinnvoll bei Zählern die ihre Werte nur langsam ändern, z.B. bei einem Gaszähler.
Mit dieser "Realtime" Einstellung wird der MQTT Server und vor allem der ESP mit OpenDTU-OnBattery dann mit unnötig vielen Werten geflutet.
Wenn man so einen Tasmota-basierten Smartmeter-Leser hat ist das abholen der Werte von OpenDTU-onBattery per http+json das sinnvollste und ressourcenschonendste bezogen auf die Rechenleistung des ESP. Dann holt der DPL nämlich die Werte wenn er sie braucht, und nicht wann sie von außen reingeballert kommen.

Wie gesagt, meine persönliche Meinung.

**Reply by Manos1966** [reply]
Hi @madmartin, ich bin auch deiner Meinung, http+json ist viel besser.
Fuer Leute die ein Homematik System haben und alle Informationen da sammeln, ist diese Information eher zutrefend.

Wo in Wiki sollte ich eine solche Information scheiben? (habe ich nie bis jetzt gemacht)

**Comment by madmartin** [comment]
Ich würde es unter Power Meter unterbringen - erstmal einen Abschnitt oder Unter-Artikel über Tasmota-based Devices anlegen und das dann hinzufügen.

https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Power-Meter

**Reply by Manos1966** [reply]
Danke. Ich habe den Link addiert.
Ich werde versuchen ein paar zusaetzliche Infos zu addieren bei anderen Themen, ich hoffe es ist O.K.
siehe https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Pylontech-battery
